### PR TITLE
PUB-1360 Pdf for Civil and Family Daily Cause List

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1360

### Change description ###

Create Pdf for Civil and Family Daily Cause list and merge the case information in email placeholder.

Image 28-07-2022 at 17 26

Image 28-07-2022 at 17 26 (1)

Image 28-07-2022 at 17 26

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotifyTest.java [hunk 1]
```diff
@@ -326,6 +328,22 @@ void testInvalidEmailForSubscriptionReturnsBadRequest() throws Exception {
...
+    @Test
+    void testValidPayloadForSubsFamilyCauseListListEmailReturnsOk() throws Exception {
+        mockMvc.perform(post(SUBSCRIPTION_URL)
+                            .content(VALID_FAMILY_CAUSE_LIST_SUBS_EMAIL)
+                            .contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk())
+            .andExpect(content().string(containsString("Subscription email successfully sent to")));
+    }
+
+    @Test
+    void testValidPayloadForSubsCivilAndFamilyCauseListListEmailReturnsOk() throws Exception {
```
Comment (KianKwa) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotifyTest.java:340]: Typo in the test name....it says "ListList" :)

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/converters/CivilAndFamilyDailyCauseListConverter.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring5.SpringTemplateEngine;
+import uk.gov.hmcts.reform.pip.publication.services.config.ThymeleafConfiguration;
+import uk.gov.hmcts.reform.pip.publication.services.service.helpers.DataManipulation;
+import uk.gov.hmcts.reform.pip.publication.services.service.helpers.Helpers;
+
+import java.util.Map;
+
+@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/converters/CivilAndFamilyDailyCauseListConverter.java:13]: There is nothing in loop in this method so do not need this PMD suppression. Probably just copying error from Danny's one

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/CivilAndFamilyDailyCauseList.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pip.publication.services.service.helpers.ArtefactSummary;
+import uk.gov.hmcts.reform.pip.publication.services.service.helpers.DataManipulation;
+
+@Service
+public class CivilAndFamilyDailyCauseList {
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/CivilAndFamilyDailyCauseList.java:11]: This class looks identical to the Family one. Is the Civil one the same too? May be just need a single class called 'DailyCauseList' to cover them all?

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotifyTest.java (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotifyTest.java]: fixed :)

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/converters/CivilAndFamilyDailyCauseListConverter.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/converters/CivilAndFamilyDailyCauseListConverter.java]: removed.

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/CivilAndFamilyDailyCauseList.java (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/CivilAndFamilyDailyCauseList.java]: yes, you are right. I created on class DailyCauseList and use in both Family and Mix list.

**Review by KianKwa — APPROVED** [review]

**Review by KianKwa — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/DailyCauseList.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    public String artefactSummaryDailyCause(String payload) throws JsonProcessingException {
+        JsonNode node = new ObjectMapper().readTree(payload);
+
+        DataManipulation.manipulatedDailyListData(node);
+
+        return this.processDailyCauseList(node);
+    }
+
+    public String processDailyCauseList(JsonNode node) {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/DailyCauseList.java:28]: Could we get away without the appendToStringBuilders in the method below and use .append? (Especially for the \n's)

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/helpers/Helpers.java [hunk 5]
```diff
@@ -9,13 +12,15 @@
 /**
  * Class for static utility methods assisting with json->html->pdf issues.
  */
+@SuppressWarnings("PMD.TooManyMethods")
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/helpers/Helpers.java:15]: I've split this out into a couple of different helper classes which managed to get rid of the annoation above in the PR for 1361 - could we do the same here?

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by furnivall — APPROVED** [review]
Looks good man nice work :D

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/DailyCauseList.java (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/artefactsummary/DailyCauseList.java:28]: I was using .append, but PMD was complaining. Now, I have refactor this function completely.

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/helpers/Helpers.java (hunk 5 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/pdf/helpers/Helpers.java]: Move all helper functions into relevant helper classes and update the test classes as well.

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**